### PR TITLE
fix(console): searching for custom key by and environment_id

### PR DESCRIPTION
**Pull request by skorda42**
## Issue

https://gravitee.atlassian.net/browse/APIM-7335
## Description

Search for custom key by key value and environment_id

## Additional context

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-pjxnfaiesg.chromatic.com)

### CODE REVIEW

File: gravitee-apim-repository/gravitee-apim-repository-jdbc/src/main/java/io/gravitee/repository/jdbc/management/JdbcApiKeyRepository.java [hunk 1]
```diff
@@ -67,6 +68,18 @@ public class JdbcApiKeyRepository extends JdbcAbstractCrudRepository<ApiKey, Str
         }
     };
 
+    private static final JdbcHelper.ChildAdder<ApiKey> CHILD_ADDER1 = (ApiKey parent, ResultSet rs) -> {
```
Comment (Okhelifi) [inline gravitee-apim-repository/gravitee-apim-repository-jdbc/src/main/java/io/gravitee/repository/jdbc/management/JdbcApiKeyRepository.java]: As it's not used can we remove it ?

File: gravitee-apim-repository/gravitee-apim-repository-jdbc/src/main/java/io/gravitee/repository/jdbc/management/JdbcApiKeyRepository.java [hunk 2]
```diff
@@ -377,6 +390,41 @@ public List<ApiKey> findByApplication(String applicationId) throws TechnicalExce
...
+                            escapeReservedWord("environment_id") +
+                            " = ?" +
+                            " order by k.updated_at desc ";
+
+//            String query = getOrm().getSelectAllSql()
+//                    + " k where k." +
+//                    escapeReservedWord("key") +
+//                    " = ? and k." +
+//                    escapeReservedWord("environment_id") +
+//                    " = ?";
```
Comment (Okhelifi) [inline gravitee-apim-repository/gravitee-apim-repository-jdbc/src/main/java/io/gravitee/repository/jdbc/management/JdbcApiKeyRepository.java]: to remove too

File: gravitee-apim-repository/gravitee-apim-repository-jdbc/src/main/java/io/gravitee/repository/jdbc/management/JdbcApiKeyRepository.java [hunk 3]
```diff
@@ -377,6 +390,41 @@ public List<ApiKey> findByApplication(String applicationId) throws TechnicalExce
         }
     }
 
+    @Override
+    public List<ApiKey> findByKeyAndEnvironmentId(String key, String environmentId) throws TechnicalException {
+        LOGGER.debug("JdbcApiKeyRepository.findBySubscription({})", subscription);
```
Comment (Okhelifi) [inline gravitee-apim-repository/gravitee-apim-repository-jdbc/src/main/java/io/gravitee/repository/jdbc/management/JdbcApiKeyRepository.java]:
```suggestion
        LOGGER.debug("JdbcApiKeyRepository.findByKeyAndEnvironmentId(****, {})", environmentId);
```

File: gravitee-apim-repository/gravitee-apim-repository-jdbc/src/main/java/io/gravitee/repository/jdbc/management/JdbcApiKeyRepository.java [hunk 4]
```diff
@@ -377,6 +390,41 @@ public List<ApiKey> findByApplication(String applicationId) throws TechnicalExce
...
+//                    " = ?";
+
+            CollatingRowMapper<ApiKey> rowMapper = new CollatingRowMapper<>(getOrm().getRowMapper(), CHILD_ADDER, "id");
+
+            jdbcTemplate.query(query, rowMapper, key, environmentId);
+
+            return rowMapper.getRows();
+        } catch (final Exception ex) {
+            LOGGER.error("Failed to find API Keys by subscription:", ex);
+            throw new TechnicalException("Failed to find API Keys by subscription", ex);
```
Comment (Okhelifi) [inline gravitee-apim-repository/gravitee-apim-repository-jdbc/src/main/java/io/gravitee/repository/jdbc/management/JdbcApiKeyRepository.java]:
```suggestion
            LOGGER.error("Failed to find API Keys by key and environment:", ex);
            throw new TechnicalException("Failed to find API Keys by key and environment", ex);
```

File: gravitee-apim-repository/gravitee-apim-repository-mongodb/src/main/java/io/gravitee/repository/mongodb/management/MongoApiKeyRepository.java [hunk 5]
```diff
@@ -125,6 +125,11 @@ public List<ApiKey> findByKey(String key) {
         return internalApiKeyRepo.findByKey(key).stream().map(this::toApiKey).collect(toList());
     }
 
+    @Override
+    public List<ApiKey> findByKeyAndEnvironmentId(String key, String environmentId) {
+        return internalApiKeyRepo.findByKeyAndEnvironmentId(key, environmentId).stream().map(this::toApiKey).collect(toList());
```
Comment (Okhelifi) [inline gravitee-apim-repository/gravitee-apim-repository-mongodb/src/main/java/io/gravitee/repository/mongodb/management/MongoApiKeyRepository.java]:
```suggestion
        return internalApiKeyRepo.findByKeyAndEnvironmentId(key, environmentId).stream().map(this::toApiKey).toList();
```

File: gravitee-apim-repository/gravitee-apim-repository-api/src/main/java/io/gravitee/repository/management/api/ApiKeyRepository.java [hunk 6]
```diff
@@ -44,6 +44,8 @@ public interface ApiKeyRepository extends FindAllRepository<ApiKey> {
      */
     List<ApiKey> findByKey(String key) throws TechnicalException;
 
+    List<ApiKey> findByKeyAndEnvironmentId(String key, String environmentId) throws TechnicalException;
```
Comment (Okhelifi) [inline gravitee-apim-repository/gravitee-apim-repository-api/src/main/java/io/gravitee/repository/management/api/ApiKeyRepository.java:47]: We miss a test in ApiKeyRepositoryTest.java I think

**Review by Okhelifi — COMMENTED** [review]
Some comments to take into account please. 
And I think that you forget to run `mvn prettier:write` the build is failing on CircleCI

File: gravitee-apim-repository/gravitee-apim-repository-api/src/main/java/io/gravitee/repository/management/api/ApiKeyRepository.java (hunk 6 above)
Comment (skorda42) [reply gravitee-apim-repository/gravitee-apim-repository-api/src/main/java/io/gravitee/repository/management/api/ApiKeyRepository.java:47]: Test added

File: gravitee-apim-repository/gravitee-apim-repository-jdbc/src/main/java/io/gravitee/repository/jdbc/management/JdbcApiKeyRepository.java [hunk 7]
```diff
@@ -377,6 +377,35 @@ public List<ApiKey> findByApplication(String applicationId) throws TechnicalExce
         }
     }
 
+    @Override
+    public List<ApiKey> findByKeyAndEnvironmentId(String key, String environmentId) throws TechnicalException {
+        LOGGER.debug("JdbcApiKeyRepository.findByKeyAndEnvironmentId({}, {})", key, environmentId);
```
Comment (Okhelifi) [inline gravitee-apim-repository/gravitee-apim-repository-jdbc/src/main/java/io/gravitee/repository/jdbc/management/JdbcApiKeyRepository.java]: maybe we should not log the key. That's why I have proposed ***** in my previous comment

File: gravitee-apim-repository/gravitee-apim-repository-test/src/test/resources/data/apikey-tests/apiKeys.json [hunk 8]
```diff
@@ -113,5 +113,29 @@
         "updatedAt": 1486771600000,
         "revoked": false,
         "federated": true
-    }
+    },
+  {
```
Comment (Okhelifi) [inline gravitee-apim-repository/gravitee-apim-repository-test/src/test/resources/data/apikey-tests/apiKeys.json]: there is a formatting issue here no ? on GH it looks like we miss a space

**Review by Okhelifi — APPROVED** [review]
Looks good for me I just have 2 small comments to take into account before merging

**Pull request merged**